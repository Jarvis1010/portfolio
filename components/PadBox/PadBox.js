import styled from "styled-components";

const allowedSpacings = (theme = {}) => {
  const defaultTheme = {
    spacing4: "0.25rem",
    spacing8: "0.5rem",
    spacing16: "1rem",
    spacing32: "2rem",
    spacing64: "4rem",
    spacing128: "8rem"
  };
  const finalTheme = { ...defaultTheme, ...theme };

  return {
    // fixed vals
    "0": "0",
    "4": finalTheme.spacing4,
    "8": finalTheme.spacing8,
    "16": finalTheme.spacing16,
    "24": "1.5rem",
    "32": finalTheme.spacing32,
    "64": finalTheme.spacing64,
    "128": finalTheme.spacing128,

    // percentage
    ".05": "5%",
    ".10": "10%",
    ".15": "15%",
    ".20": "20%"
  };
};

const handlePaddingFor = pos => ({ size = "32", ...props }) =>
  (props[pos] && allowedSpacings()[props[pos]]) || allowedSpacings()[size];

const PadBox = styled.div`
  width: 100%;
  height: 100%;
  padding-bottom: ${handlePaddingFor("bottom")};
  padding-left: ${handlePaddingFor("left")};
  padding-top: ${handlePaddingFor("top")};
  padding-right: ${handlePaddingFor("right")};
`;

export default PadBox;
