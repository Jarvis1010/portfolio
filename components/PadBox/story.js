import React from "react";
import { storiesOf } from "@storybook/react";
import PadBox from "./PadBox";

const filler = (
  <div style={{ border: "1px solid black", background: "lightgrey" }}>
    Content
  </div>
);

storiesOf("PadBox", module)
  .add(
    "PadBox updates all sides with default size attribute when no props passed",
    () => <PadBox style={{ border: "1px solid black" }}>{filler}</PadBox>
  )
  .add("PadBox updates all sides with size attribute", () => (
    <PadBox size="128" style={{ border: "1px solid black" }}>
      {filler}
    </PadBox>
  ))
  .add("PadBox update top with top attribute ", () => (
    <PadBox top="128" style={{ border: "1px solid black" }}>
      {filler}
    </PadBox>
  ))
  .add(
    "PadBox updates all sides except top with size attribute and top is set to different size",
    () => (
      <PadBox top="0" size="128" style={{ border: "1px solid black" }}>
        {filler}
      </PadBox>
    )
  );
