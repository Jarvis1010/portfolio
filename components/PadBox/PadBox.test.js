import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import PadBox from "./PadBox";

describe("PadBox Tests", () => {
  it("renders for all the correct sizes", () => {
    const sizes = [
      "0",
      "4",
      "8",
      "16",
      "24",
      "32",
      "64",
      "128",
      ".05",
      ".10",
      ".15",
      ".20",
      null
    ];
    sizes.forEach(size => {
      const wrapper = shallow(<PadBox size={size} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  it("adds padding to only bottom", () => {
    const wrapper = shallow(<PadBox bottom="0" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("adds padding to only top", () => {
    const wrapper = shallow(<PadBox top="0" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("adds padding to only left", () => {
    const wrapper = shallow(<PadBox left="0" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("adds padding to only right", () => {
    const wrapper = shallow(<PadBox right="0" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("adds padding to only right and size value for the rest of the sides", () => {
    const wrapper = shallow(<PadBox right="0" size=".20" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
