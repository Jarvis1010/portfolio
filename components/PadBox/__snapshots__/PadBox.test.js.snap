// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PadBox Tests adds padding to only bottom 1`] = `
.c0 {
  width: 100%;
  padding-bottom: 0;
  padding-left: 2rem;
  padding-top: 2rem;
  padding-right: 2rem;
}

<div
  className="c0"
/>
`;

exports[`PadBox Tests adds padding to only left 1`] = `
.c0 {
  width: 100%;
  padding-bottom: 2rem;
  padding-left: 0;
  padding-top: 2rem;
  padding-right: 2rem;
}

<div
  className="c0"
/>
`;

exports[`PadBox Tests adds padding to only right 1`] = `
.c0 {
  width: 100%;
  padding-bottom: 2rem;
  padding-left: 2rem;
  padding-top: 2rem;
  padding-right: 0;
}

<div
  className="c0"
/>
`;

exports[`PadBox Tests adds padding to only right and size value for the rest of the sides 1`] = `
.c0 {
  width: 100%;
  padding-bottom: 20%;
  padding-left: 20%;
  padding-top: 20%;
  padding-right: 0;
}

<div
  className="c0"
  size=".20"
/>
`;

exports[`PadBox Tests adds padding to only top 1`] = `
.c0 {
  width: 100%;
  padding-bottom: 2rem;
  padding-left: 2rem;
  padding-top: 0;
  padding-right: 2rem;
}

<div
  className="c0"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 1`] = `
.c0 {
  width: 100%;
  padding-bottom: 0;
  padding-left: 0;
  padding-top: 0;
  padding-right: 0;
}

<div
  className="c0"
  size="0"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 2`] = `
.c0 {
  width: 100%;
  padding-bottom: 0.25rem;
  padding-left: 0.25rem;
  padding-top: 0.25rem;
  padding-right: 0.25rem;
}

<div
  className="c0"
  size="4"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 3`] = `
.c0 {
  width: 100%;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-top: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
  size="8"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 4`] = `
.c0 {
  width: 100%;
  padding-bottom: 1rem;
  padding-left: 1rem;
  padding-top: 1rem;
  padding-right: 1rem;
}

<div
  className="c0"
  size="16"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 5`] = `
.c0 {
  width: 100%;
  padding-bottom: 1.5rem;
  padding-left: 1.5rem;
  padding-top: 1.5rem;
  padding-right: 1.5rem;
}

<div
  className="c0"
  size="24"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 6`] = `
.c0 {
  width: 100%;
  padding-bottom: 2rem;
  padding-left: 2rem;
  padding-top: 2rem;
  padding-right: 2rem;
}

<div
  className="c0"
  size="32"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 7`] = `
.c0 {
  width: 100%;
  padding-bottom: 4rem;
  padding-left: 4rem;
  padding-top: 4rem;
  padding-right: 4rem;
}

<div
  className="c0"
  size="64"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 8`] = `
.c0 {
  width: 100%;
  padding-bottom: 8rem;
  padding-left: 8rem;
  padding-top: 8rem;
  padding-right: 8rem;
}

<div
  className="c0"
  size="128"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 9`] = `
.c0 {
  width: 100%;
  padding-bottom: 5%;
  padding-left: 5%;
  padding-top: 5%;
  padding-right: 5%;
}

<div
  className="c0"
  size=".05"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 10`] = `
.c0 {
  width: 100%;
  padding-bottom: 10%;
  padding-left: 10%;
  padding-top: 10%;
  padding-right: 10%;
}

<div
  className="c0"
  size=".10"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 11`] = `
.c0 {
  width: 100%;
  padding-bottom: 15%;
  padding-left: 15%;
  padding-top: 15%;
  padding-right: 15%;
}

<div
  className="c0"
  size=".15"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 12`] = `
.c0 {
  width: 100%;
  padding-bottom: 20%;
  padding-left: 20%;
  padding-top: 20%;
  padding-right: 20%;
}

<div
  className="c0"
  size=".20"
/>
`;

exports[`PadBox Tests renders for all the correct sizes 13`] = `
.c0 {
  width: 100%;
}

<div
  className="c0"
  size={null}
/>
`;
